packages/cadl-python/src/emitter.ts

```diff
@@ -187,11 +187,9 @@ function getEffectiveSchemaType(program: Program, type: Model): Model {
         return !(headerInfo || queryInfo || pathInfo || statusCodeinfo);
     }
 
-    if (type.kind === "Model" && !type.name) {
-        const effective = getEffectiveModelType(program, type, isSchemaProperty);
```

Comment:
the check is unproper since we have checked its kind and name before step into the function:
https://github.com/Azure/autorest.python/blob/a6ca7f27e1d3d91f2035c308f2e9399c40e758d4/packages/cadl-python/src/emitter.ts#L434-L436.
So I think we can add test in cadl-ranch for this kind of scenario.
It may  be added  unguardedly by https://github.com/Azure/autorest.python/pull/1439/files#diff-fdccd5322f2233513a40525093d9ef9a33c2af64250f90c79f2dce2b89e2a69f:
![image](https://user-images.githubusercontent.com/70930885/192448638-ac131005-6fa4-4ae5-bf0d-05361480685a.png)